config/registry_image_pruner/image_pruner/prune_images.py

```diff
@@ -1,53 +1,104 @@
 import argparse
-import os
+import itertools
+import json
 import logging
+import os
 import re
-import requests
+
+from collections.abc import Iterator
+from http.client import HTTPResponse
+from typing import Any, Dict, List
+from urllib.parse import urlencode
+from urllib.request import Request, urlopen
 
 logging.basicConfig(
     format="%(asctime)s - %(levelname)s - %(message)s", level=logging.INFO
 )
 LOGGER = logging.getLogger(__name__)
 QUAY_API_URL = "https://quay.io/api/v1"
 
-PROCESSED_REPOS = 0
+processed_repos_counter = itertools.count()
+
+
+ImageRepo = Dict[str, Any]
+
+
+def get_quay_repo(quay_token: str, namespace: str, name: str) -> ImageRepo:
+    api_url = f"{QUAY_API_URL}/repository/{namespace}/{name}"
+    request = Request(api_url, headers={
+        "Authorization": f"Bearer {quay_token}",
+    })
+    resp: HTTPResponse
+    with urlopen(request) as resp:
+        if resp.status != 200:
+            raise RuntimeError(resp.reason)
+        return json.loads(resp.read())
+
+
+def delete_image_repo(quay_token: str, namespace: str, name: str, tag: str) -> None:
+    api_url = f"{QUAY_API_URL}/repository/{namespace}/{name}/tag/{tag}"
+    request = Request(api_url, method="DELETE", headers={
+        "Authorization": f"Bearer {quay_token}",
+    })
+    resp: HTTPResponse
+    with urlopen(request) as resp:
+        if resp.status != 200 and resp.status != 204:
+            raise RuntimeError(resp.reason)
 
 
-def remove_images(images, session, repository, dry_run=False):
+def remove_images(images: Dict[str, Any], quay_token: str, namespace: str, name: str, dry_run: bool = False) -> None:
     image_digests = [image["manifest_digest"] for image in images.values()]
-    for image in images:
-        # attribute or sbom image
-        if image.endswith(".att") or image.endswith(".sbom"):
-            sha = re.search("sha256-(.*)(.sbom|.att)", image).group(1)
-            if f"sha256:{sha}" not in image_digests:
+    image_regex = re.compile(r"^sha256-([0-9a-f]+)(\.sbom|\.att)$")
+    for tag in images:
+        # attestation or sbom image
+        if (match := image_regex.match(tag)) is not None:
```

Comment:
nitpick: with the rename of `image` -> `tag`, the regex should probably be called `tag_regex`

Reply:
since we decided to rename, when iterating 
```
for tag in images
```
I think we are iterating through all the tags for certain image (in quay) so should be something like for tag in tags
also, then we could change the name of the method as well.

So I feel like we should not mix those names if they refer to the same thing

Reply:
@mkosiarc PTAL. More replacements are made.